common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java

```diff
@@ -0,0 +1,270 @@
+import com.wynntils.features.user.overlays.map.PointerType;
+import com.wynntils.mc.objects.CommonColors;
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.KeyboardUtils;
+import com.wynntils.utils.MathUtils;
+import com.wynntils.wynn.model.CompassModel;
+import net.minecraft.client.KeyMapping;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.resources.sounds.SimpleSoundInstance;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.sounds.SoundEvents;
+import org.lwjgl.glfw.GLFW;
+
+public class MainMapScreen extends Screen {
+    private static final float SCREEN_SIDE_OFFSET = 10;
+    private static final float BORDER_OFFSET = 7;
+
+    private boolean holdingMapKey = false;
+
+    private float renderWidth;
+    private float renderHeight;
+    private float renderX;
+    private float renderY;
+
+    private float renderedBorderXOffset;
+    private float renderedBorderYOffset;
+
+    private float mapWidth;
+    private float mapHeight;
+    private float centerX;
+    private float centerZ;
+
+    private float mapCenterX;
+    private float mapCenterZ;
+
+    // Zoom is the scaling of the map. The bigger the zoom, the less detailed the map becomes.
+    private static final float MIN_ZOOM = 5.0f;
+    private static final float MAX_ZOOM = 0.3f;
+    private static final float ZOOM_FACTOR = 0.4f;
+    private float currentZoom = 1f;
+
+    private boolean dragging = false;
+    private double lastMouseX = 0;
+    private double lastMouseY = 0;
+
+    public MainMapScreen() {
+        super(new TextComponent("Main Map"));
+        centerMapAroundPlayer();
+    }
+
+    @Override
+    protected void init() {
+        // FIXME: Figure out a way to not need this.
+        //        At the moment, this is needed for Minecraft not to forget we hold keys when we open the GUI...
+        KeyMapping.set(
+                McUtils.mc().options.keyUp.key,
+                KeyboardUtils.isKeyDown(McUtils.mc().options.keyUp.key.getValue()));
+        KeyMapping.set(
+                McUtils.mc().options.keyDown.key,
+                KeyboardUtils.isKeyDown(McUtils.mc().options.keyDown.key.getValue()));
+        KeyMapping.set(
+                McUtils.mc().options.keyLeft.key,
+                KeyboardUtils.isKeyDown(McUtils.mc().options.keyUp.key.getValue()));
+        KeyMapping.set(
+                McUtils.mc().options.keyRight.key,
+                KeyboardUtils.isKeyDown(McUtils.mc().options.keyRight.key.getValue()));
+        KeyMapping.set(
+                McUtils.mc().options.keyJump.key,
+                KeyboardUtils.isKeyDown(McUtils.mc().options.keyUp.key.getValue()));
+        KeyMapping.set(
+                McUtils.mc().options.keyShift.key,
+                KeyboardUtils.isKeyDown(McUtils.mc().options.keyShift.key.getValue()));
```

Comment:
What does this do?

Reply:
When opening a map, you stop running/shifting/etc. This makes it so you don't (as it is done in legacy)

Reply:
Why do we do that? Can you like run and look at the map at the same time in Legacy? 😮

Reply:
Yes